src/main/java/domain/WinningResult.java

```diff
@@ -1,17 +1,55 @@
 package domain;
 
+import domain.card.Cards;
+import domain.card.Score;
+
+import java.util.Arrays;
+import java.util.function.BiPredicate;
+
 public enum WinningResult {
-    WIN("승"),
-    DRAW("무"),
-    LOSE("패");
+    WIN(WinningResult::isWin, 1),
+    DRAW(WinningResult::isDraw, 0),
+    LOSE(WinningResult::isLose, -1),
+    BLACKJACK(WinningResult::isBlackjack, 1.5);
+
+    private final BiPredicate<Cards, Cards> compare;
+    private final double profitRatio;
+
+    WinningResult(BiPredicate<Cards, Cards> compare,
+                  double profitRatio) {
+        this.compare = compare;
+        this.profitRatio = profitRatio;
+    }
+
+    public static WinningResult of(Cards playerCards, Cards dealerCards) {
+        return Arrays.stream(values())
+                .filter(outcome -> outcome.compare.test(playerCards, dealerCards))
+                .findAny()
+                .orElseThrow(IllegalArgumentException::new);
+    }
+
+    private static boolean isWin(Cards playerCards, Cards dealerCards) {
+        if (playerCards.isBust() || playerCards.isBlackJack()) {
+            return false;
+        }
+        return dealerCards.isBust() || playerCards.getScore().isMoreThan(dealerCards.getScore());
```

Comment:
```suggestion
        return dealerCards.isBust() || playerCards.isMoreThanScore(dealerCards);
```

와 같이 getter를 제거한다면 어떠한 장점이 있을지 고민해보아요.